File: LeetCode/5회차/B07-1-Two Sum.py
from typing import List
from bisect import bisect_left


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        temp = list((nums[i], i) for i in range(len(nums)))
        temp.sort()
        for i, (num, idx) in enumerate(temp):
            j = bisect_left(nums, target - nums[i], i + 1)
            while i < j < len(temp) and num + temp[j][0] == target:
                if i < j < len(temp) and num + temp[j][0] == target:
                    return [idx, temp[j][1]]
        return [0, 0]


print(Solution().twoSum([3,2,4],
6))

File: LeetCode/JavaScript/B07-1-Two Sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const indices = {};
  let res = [];
  for (let i = 0; i < nums.length; i++) {
    if (indices[target - nums[i]] > -1) {
      return [i, indices[target - nums[i]]];
    }
    indices[nums[i]] = i;
  }
};

console.log(twoSum([2, 7, 11, 15], 9));
